core/src/main/java/io/grpc/internal/DelayedClientTransport.java

```diff
@@ -307,8 +307,9 @@ boolean isInBackoffPeriod() {
    */
   void startBackoff(final Status status) {
     synchronized (lock) {
-      Preconditions.checkState(!inBackoffPeriod);
-      inBackoffPeriod = true;
+      Preconditions.checkState(backoffStatus == null);
+      backoffStatus = Status.UNAVAILABLE.withDescription("Channel in TRANSIENT_FAILURE state")
+          .withCause(status.asRuntimeException());
```

Comment:
oh right, just saw it is used in `newStream` method. 
I think we can verify this somewhere in the DelayedTransportTest or
in the testcase 
`TransportSetTest#verifyFailFastAndNonFailFastBehaviors()`
between line# 389-392
```
    // Create a new fail fast stream.
    delayedTransport.newStream(method, headers, failFastCallOptions);  
```
// verify it is a failing stream with back-off information
```
    // Verify it is not queued.
    assertEquals(pendingStreamsCount, delayedTransport.getPendingStreamsCount());
```